Agents/BMSUpdateAgent/BMSUpdateAgent/src/main/java/uk/ac/cam/cares/jps/agent/bmsupdate/BMSUpdateAgentLauncher.java

```diff
@@ -107,66 +156,163 @@ public JSONObject processRequestParameters(JSONObject requestParams) {
+            BMSUpdateAgent bmsUpdateAgent = new BMSUpdateAgent();
+            try {
+                initSparqlProperties(clientPropertiesFile);
+            } catch (IOException e) {
+                throw new JPSRuntimeException("Unable to read the client properties file.", e);
+            }
+            rsClient = new RemoteStoreClient(sparqlQueryEndpoint, sparqlUpdateEndpoint);
+            if (sparqlUsername != null && sparqlPassword != null) {
+                rsClient.setUser(sparqlUsername);
+                rsClient.setPassword(sparqlPassword);
+            }
+            //query for Bacnet Device ID linked to data IRI
+            String deviceId = bmsUpdateAgent.getDeviceId(dataIRI, rsClient);
+            //query for Bacnet Object ID linked to data IRI
+            String objectId = bmsUpdateAgent.getObjectId(dataIRI, rsClient);
+            //send request via API
+            try {
+                BMSWacnetAPIConnector wacnetAPIConnector = new BMSWacnetAPIConnector(System.getenv(ENV_WACNET_API_PROPERTIES));
+                result = wacnetAPIConnector.writePresentValue(deviceId, objectId, value);
+                return result;
+            } catch (Exception e) {
+                throw new JPSRuntimeException("Unable to write present value to Bacnet Object: " + objectId, e);
+            }
+        } else if (requestParams.has(KEY_BACNETDEVICEID) && requestParams.has(KEY_BACNETOBJECTID) && requestParams.has(KEY_VALUE)) {
+            //retrieve Bacnet Device ID from request parameters
+            String deviceId = requestParams.getString(KEY_BACNETDEVICEID);
+            //retrieve Bacnet Object ID from request parameters
+            String objectId = requestParams.getString(KEY_BACNETOBJECTID);
+            Double value = requestParams.getDouble(KEY_VALUE);
+            //send request via API
+            try {
+                BMSWacnetAPIConnector wacnetAPIConnector = new BMSWacnetAPIConnector(System.getenv(ENV_WACNET_API_PROPERTIES));
+                result = wacnetAPIConnector.writePresentValue(deviceId, objectId, value);
+                return result;
+            } catch (IOException e) {
+                throw new JPSRuntimeException("Unable to write present value to Bacnet Object: " + objectId, e);
+            }
+        } else {
+            throw new JPSRuntimeException("missing keys and values in request!");
         }
+    }
 
-        if (!requestParams.has(KEY_CLIENT_PROPERTIES)) {
-            LOGGER.error(KEY_CLIENT_PROPERTIES + " is missing");
-            return false;
+    /**
+     * execute update triples route
+     * @param requestParams Request parameters parsed from the body of POST request
+     * @return results of executed route
+     */
+    public JSONObject executeUpdateTriples(JSONObject requestParams) {
+        JSONObject result = new JSONObject();
+        LOGGER.info("The size is " + requestParams.length());
```

Comment:
The size of requestParams is always 1. Is this intended to check checks.length()?

Reply:
Oh yes. I added that for debugging initially and forgot to remove it, my bad. I have removed that line now.